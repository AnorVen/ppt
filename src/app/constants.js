import { CustomRow } from '@/components/custom-row';

import { CustomFilter } from '@/components/custom-filter';
import { base_themes, master_themes } from '@/constants';

export const FORM_NAME = 'form-ppt'
export  const COLUMNS = [
	{
		name: 'Тип Курса',
		id: 'type',
		customHeaderComponent: CustomFilter,
		filterName: 'withCourseType',
		alignLeft: false
	},
	{
		name: 'Модуль',
		id: 'module',
		customHeaderComponent: CustomFilter,
		filterName: 'withModule',
		alignLeft: false
	},
	{
		name: 'Город',
		id: 'city',
		customHeaderComponent: CustomFilter,
		filterName: 'withCity',
		alignLeft: true
	},
	{
		name: 'Даты',
		id: 'dates',
		customHeaderComponent: CustomFilter,
		filterName: 'withDates',
		alignLeft: true
	},
	{
		name: 'Тренер',
		id: 'trainer',
		customHeaderComponent: CustomFilter,
		filterName: 'withTrainer',
		alignLeft: true
	},
	{
		name: 'Организатор',
		id: 'organizer',
		customComponent: CustomRow,
	},
	{
		name: 'Описание',
		id: 'description',
		customComponent: CustomRow,
	},
];

export const filterNamesToColumns = COLUMNS.reduce((acc, val)=>{
	if(val.filterName){
		acc[val.filterName] = val.id
	}
	return acc
}, {})

export const TEXT_FIELD_PROPS = {
	formats: [
		'header',
		'bold',
		'italic',
		'underline',
		'strike',
		'blockquote',
		'list',
		'bullet',
		'indent',
		'link',
		'image',
		'align',
	],
	modules: {
		toolbar: [
			[{ header: [1, 2, false] }],
			['bold', 'italic', 'underline', 'strike', 'blockquote'],
			[{ list: 'ordered' }, { list: 'bullet' }, { indent: '-1' }, { indent: '+1' }],
			[{ align: [] }],
			['link'],
			['clean'],
		],
	},
};



export const basic_course = Object.keys(base_themes).reduce((acc,key, index )=>{
	acc.push({
		module_number: index + 1,
		trainer: null,
		dates: [],
		count: 24,
		them: key,
	})
	return acc
}, [])
export const master_course = Object.keys(master_themes).reduce((acc,key, index )=>{
	acc.push({
		module_number: index + 1,
		trainer: null,
		dates: [],
		count: 24,
		them: key,
	})
	return acc
}, [])