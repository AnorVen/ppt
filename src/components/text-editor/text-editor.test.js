/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-props-no-spreading */
import { render, screen, act, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { TextEditorWrapper } from '.';
import { TEXT_FIELD_PROPS } from '../../constants';

const mockReactQuillComponent = jest.fn();

jest.mock('react-quill', () => ({
	__esModule: true,
	default: props => {
		mockReactQuillComponent(props);
		return (
			<div data-testid="react-quill" onClick={props.onChange}>
				{props.value}
			</div>
		);
	},
}));

jest.mock('../../constants', () => ({
	TEXT_FIELD_PROPS: {
		formats: [
			'header',
			'bold',
			'italic',
			'underline',
			'strike',
			'blockquote',
			'list',
			'bullet',
			'indent',
			'link',
			'image',
			'align',
		],
		modules: {
			toolbar: [
				[{ header: [1, 2, false] }],
				['bold', 'italic', 'underline', 'strike', 'blockquote'],
				[{ list: 'ordered' }, { list: 'bullet' }, { indent: '-1' }, { indent: '+1' }],
				[{ align: [] }],
				['link'],
				['clean'],
			],
		},
	},
}));

const props = {
	onSetMainTextValue: jest.fn(),
	mainTextValue: 'components',
};

describe('TextEditorWrapper', () => {
	it('should  render TextEditorWrapper', () => {
		render(<TextEditorWrapper {...props} />);
		expect(screen.queryByTestId('react-quill')).toBeInTheDocument();
		expect(mockReactQuillComponent).toBeCalledWith({
			formats: TEXT_FIELD_PROPS.formats,
			modules: TEXT_FIELD_PROPS.modules,
			onChange: expect.any(Function),
			value: props.mainTextValue,
		});
	});

	it('should call onSetMainTextValue', async () => {
		render(<TextEditorWrapper {...props} />);

		act(() => {
			userEvent.click(screen.queryByTestId('react-quill'));
		});
		await waitFor(() => {
			expect(props.onSetMainTextValue).toBeCalled();
		});
	});

	test('snapshot test TextEditorWrapper', async () => {
		const component = renderer.create(<TextEditorWrapper {...props} />).toJSON();
		expect(component).toMatchSnapshot();
	});
});
