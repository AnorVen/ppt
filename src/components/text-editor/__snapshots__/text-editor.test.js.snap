// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditorWrapper snapshot test TextEditorWrapper 1`] = `
<div
  data-testid="react-quill"
  onClick={[Function]}
>
  components
</div>
`;
